projector/binding.go

```diff
@@ -54,35 +58,95 @@ func New(mappingSource MappingSource) ServiceBindingProjector {
 }
 
 func (p *serviceBindingProjector) Project(ctx context.Context, binding *servicebindingv1beta1.ServiceBinding, workload runtime.Object) error {
-	mapping, err := p.mappingSource.LookupMapping(ctx, workload)
+	ctx, resourceMapping, version, err := p.lookupClusterMapping(ctx, workload)
 	if err != nil {
 		return err
 	}
-	mpt, err := NewMetaPodTemplate(ctx, workload, mapping)
+
+	// rather than attempt to merge an existing binding, unproject it
+	if err := p.Unproject(ctx, binding, workload); err != nil {
+		return err
+	}
+
+	versionMapping := MappingVersion(version, resourceMapping)
+	mpt, err := NewMetaPodTemplate(ctx, workload, versionMapping)
 	if err != nil {
 		return err
 	}
 	p.project(binding, mpt)
-	return mpt.WriteToWorkload(ctx)
+
+	if p.secretName(binding) != "" {
+		if err := p.stashLocalMapping(binding, mpt, resourceMapping); err != nil {
+			return err
+		}
+	}
+	if err := mpt.WriteToWorkload(ctx); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Why was this change made?
```suggestion
	return mpt.WriteToWorkload(ctx)
```

Reply:
Functionally identical. During implementation I was playing around with the order of operations so it wasn't always the last line of the function.

Style wise, I've started to add the explicit err check rather than rely on the function signatures being identical, but it's not a strongly held opinion.